File: CreationalPatterns/Factory/C#/Vehicles/Extensions/StringExtensions.cs
using System.Linq;

namespace Vehicles.Extensions
{
    public static class StringExtensions
    {
        public static string SplitPascalCaseString(this string pascalCaseString)
        {
            var modifiedCharacters = pascalCaseString
                .SelectMany((c, i) =>
                    i != 0 && char.IsUpper(c) && !char.IsUpper(pascalCaseString[i - 1])
                        ? new char[] { ' ', c }
                        : new char[] { c });

            return new string(modifiedCharacters.ToArray());
        }
    }
}

File: CreationalPatterns/Factory/Java/Vehicles/src/extensions/StringExtensions.java
package extensions;

public class StringExtensions {
    public static String splitPascalCaseString(String pascalCaseString) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < pascalCaseString.length(); i++) {
            char currentChar = pascalCaseString.charAt(i);

            if (i != 0 && Character.isUpperCase(currentChar)
                    && !Character.isUpperCase(pascalCaseString.charAt(i - 1))) {
                result.append(' ');
            }

            result.append(currentChar);
        }

        return result.toString();
    }
}